### PR TITLE
Fix error after clean up context

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #49

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

File: library/spdm_responder_conformance_test_lib/spdm_responder_test_2_capabilities.c [hunk 1]
```diff
@@ -623,13 +623,14 @@ void spdm_test_case_capabilities_invalid_request (void *test_context)
                  (SPDM_TEST_VERSION_MASK_V11 | SPDM_TEST_VERSION_MASK_V12)) != 0) {
                 common_test_record_test_message ("test v11 flags - 0x%08x\n",
                                                  invalid_flags_v11[index]);
-                if ((test_buffer->support_version_bitmask & SPDM_TEST_VERSION_MASK_V12) != 0) {
-                    version = SPDM_MESSAGE_VERSION_12;
-                    spdm_request_size = sizeof(spdm_request);
-                } else {
+                if ((test_buffer->support_version_bitmask & SPDM_TEST_VERSION_MASK_V11) != 0) {
```
Comment (jyao1) [inline library/spdm_responder_conformance_test_lib/spdm_responder_test_2_capabilities.c:626]: why you need reverse the order of V12 and V11?

File: library/spdm_responder_conformance_test_lib/spdm_responder_test_6_challenge_auth.c [hunk 2]
```diff
@@ -553,12 +553,27 @@ void spdm_test_case_challenge_auth_success_10_12 (void *test_context, uint8_t ve
...
-                spdm_context, true, signature_ptr, test_buffer->signature_size);
-            if (result) {
-                test_result = COMMON_TEST_RESULT_PASS;
+                     spdm_context, true, signature_ptr, test_buffer->signature_size);
+            if ((case_id == SPDM_RESPONDER_TEST_CASE_CHALLENGE_AUTH_SUCCESS_10_A1B2C1) ||
+                (case_id == SPDM_RESPONDER_TEST_CASE_CHALLENGE_AUTH_SUCCESS_10_A1B3C1) ||
+                (case_id == SPDM_RESPONDER_TEST_CASE_CHALLENGE_AUTH_SUCCESS_12_A1B2C1) ||
+                (case_id == SPDM_RESPONDER_TEST_CASE_CHALLENGE_AUTH_SUCCESS_12_A1B3C1)) {
+                /* these case will verify signature failed, because they don't call get_cert
+                 * and spdm_context->connection_info.peer_used_cert_chain is NULL.
```
Comment (jyao1) [inline library/spdm_responder_conformance_test_lib/spdm_responder_test_6_challenge_auth.c:564]: I think the right fix is to collect certificate and cache it.

**Comment by jyao1** [comment]
In general, please split the fix to 4 different PR. Each PR resolves one problem.

File: library/spdm_responder_conformance_test_lib/spdm_responder_test_2_capabilities.c (hunk 1 above)
Comment (Wenxing-hou) [reply library/spdm_responder_conformance_test_lib/spdm_responder_test_2_capabilities.c:626]: Thanks. I have changed the code to only check the `(mut_auth_cap == 1) && (encap_cap == 0)` for version1.1 by the PR: #53

File: library/spdm_responder_conformance_test_lib/spdm_responder_test_6_challenge_auth.c (hunk 2 above)
Comment (Wenxing-hou) [reply library/spdm_responder_conformance_test_lib/spdm_responder_test_6_challenge_auth.c:564]: I have submitted the PR: #54 to fix it.

**Comment by Wenxing-hou** [comment]
> In general, please split the fix to 4 different PR. Each PR resolves one problem.

I have submitted 4 PR to fix the issue. And when the 4 PR is merged, this draft PR 52 can be closed.

**Pull request closed without merging**